### PR TITLE
Delayed emit messages

### PR DESCRIPTION
**Pull request by shanefagan**
I went with a thread for the delayed emits so it doesn't block. I guess we would have to give the warning that delayed emits should be used carefully to avoid race conditions.

### CODE REVIEW

**Comment by shanefagan** [comment]
Fixed the conflicts, should be fine to review again

File: sample_rules/simple0_delay.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- condition: transmission_gear == 'reverse'
+  emit:
+    signal: car.backup
+    value: true
+    delay: 100
```
Comment (treitter) [inline sample_rules/simple0_delay.yaml]: Change this delay to 1000 so it's readily observed when loading this config file manually; 100ms delays are both hard to notice on its own and easily lost in any incidental system load.

**Review by treitter — CHANGES_REQUESTED** [review]
Otherwise, looks good. Please make this change and re-submit.

File: sample_rules/simple0_delay.yaml (hunk 1 above)
Comment (shanefagan) [reply sample_rules/simple0_delay.yaml]: Done

**Pull request merged**